hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -174,6 +185,16 @@ private HoodieWriteConfig createMetadataWriteConfig(HoodieWriteConfig writeConfi
     int minCommitsToKeep = Math.max(writeConfig.getMetadataMinCommitsToKeep(), writeConfig.getMinCommitsToKeep());
     int maxCommitsToKeep = Math.max(writeConfig.getMetadataMaxCommitsToKeep(), writeConfig.getMaxCommitsToKeep());
 
+    // File groups in each partitions are fixed at creation time and we do not want them to be split into muliple files
+    // ever. Hence we use a very large basefile size in metadata table. The actual size of the HFiles created will
+    // eventually depend on the number of file groups seleted for each partition (See estimateFileGroupCount function)
+    final long maxHFileSizeBytes = 10 * 1024 * 1024 * 1024L; // 10GB
```

Comment:
Does the mapping of row key to file group breaks down if the HFile actually gets to this size and a new base file gets created? Should we just disable the max file check here?